src/sail_json_backend/json.ml

```diff
@@ -231,20 +232,62 @@ let parse_encdec i mc format =
       end
   | _ -> assert false
 
+let rec filter_non_operands components =
+  match components with
+  | [] -> []
+  | hd :: tl when String.trim hd = "spc" -> tl
+  | _ -> filter_non_operands (List.tl components)
+
+let rec extract_operands k filtered_components =
+  match filtered_components with
+  | [] -> []
+  | hd :: tl ->
+      if Str.string_match (Str.regexp ".+(\\(.*\\))") hd 0 then (
+        let operand = Str.matched_group 1 hd in
+        let elements = Str.split (Str.regexp ",") operand in
+        try
+          let inputl = Hashtbl.find inputs k in
+          let filtered_elements = List.filter (fun element -> List.mem element inputl) elements in
+          filtered_elements @ extract_operands k tl
+        with Not_found -> extract_operands k tl
```

Comment:
Sorry, a little bit "peeling the onion" here. I tend to avoid `try` blocks if there is a reasonable and straightforward alternative, and there's a bit of redundancy in those last two lines. What do you think about this alternative?
```
+          let filtered_elements = match Hashtbl.find_opt inputs k with
+              | None -> []
+              | Some inputl -> List.filter (fun element -> List.mem element inputl) elements
+          in filtered_elements @ extract_operands k tl
```

Reply:
Much better, I'll adjust and squash.